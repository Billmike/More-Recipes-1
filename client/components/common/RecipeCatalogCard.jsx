import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import RecipeImage from '../common/RecipeImage';

const propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    category: PropTypes.string,
    description: PropTypes.string,
    preparationTime: PropTypes.number,
    ingredients: PropTypes.string,
    directions: PropTypes.string,
  }),
};

const defaultProps = {
  recipe: undefined,
};

/**
 * Recipe catalog card component
 *
 * @param {object} props - The properties passed to the component
 *
 * @returns {object} React element
 */
const RecipeCatalogCard = (props) => {
  const { recipe } = props;
  const { id, title, recipeImage, description, views, upvotes, downvotes,
    favorites, User: { id: userId, fullName } } = recipe;

  return (
    <div className="col s12 m4 l3">
      <div id="recipes" className="card large">
        <Link to={`recipes/${id}`}>
          <span className="card-title">{title}</span>
        </Link>
        <div className="card-image">
          <Link to={`recipes/${id}`}>
            <RecipeImage title={title} recipeImage={recipeImage} />
          </Link>
        </div>
        <div className="card-content">
          <p>{description}</p>
        </div>
        <div className="card-action">

          <p id="owner">
            <label htmlFor="owner">Recipe by: </label>
            <Link to={`/users/${userId}/dashboard`}>
              <span className="recipe-owner">{fullName}</span>
            </Link>
          </p>

          <ul className="center-align">
            <li id="views"><i className="material-icons tiny">visibility</i>
              {views}
            </li>
            <li id="favorites"><i className="material-icons tiny">favorite</i>
              {favorites}
            </li>
            <li id="upvotes"><i className="material-icons tiny">thumb_up</i>
              {upvotes}
            </li>
            <li id="downvotes"><i className="material-icons tiny">thumb_down</i>
              {downvotes}
            </li>
          </ul>
        </div>
      </div>
    </div>
  );
};

RecipeCatalogCard.propTypes = propTypes;
RecipeCatalogCard.defaultProps = defaultProps;

export default RecipeCatalogCard;
