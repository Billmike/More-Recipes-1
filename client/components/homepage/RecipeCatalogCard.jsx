import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import common from '../common';

const { RecipeImage } = common;

const RecipeCatalogCard = (props) => {
  const { recipe } = props;
  return (
    <div className="col s12 m6 l4">
      <div id="recipes" className="card large">
        <Link to={`recipes/${recipe.id}`}>
          <span className="card-title">{recipe.title}</span>
        </Link>
        <div className="card-image">
          <Link to={`recipes/${recipe.id}`}>
            <RecipeImage />
          </Link>
        </div>
        <div className="card-content">
          <p>{recipe.description}</p>
        </div>
        <div className="card-action">
          <p id="owner">
            <label htmlFor="owner">Recipe by: </label> John Stew
          </p>
          <ul className="center-align">
            <li id="views"><i className="material-icons tiny">visibility</i>
              {recipe.views}
            </li>
            <li id="favorites"><i className="material-icons tiny">favorite</i>
              20
            </li>
            <li id="upvotes"><i className="material-icons tiny">thumb_up</i>
              {recipe.upvotes}
            </li>
            <li id="downvotes"><i className="material-icons tiny">thumb_down</i>
              {recipe.downvotes}
            </li>
          </ul>
        </div>
      </div>
    </div>
  );
};

RecipeCatalogCard.propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    category: PropTypes.string,
    description: PropTypes.string,
    preparationTime: PropTypes.number,
    ingredients: PropTypes.string,
    directions: PropTypes.string,
  }),
};

RecipeCatalogCard.defaultProps = {
  recipe: undefined,
};

export default RecipeCatalogCard;
