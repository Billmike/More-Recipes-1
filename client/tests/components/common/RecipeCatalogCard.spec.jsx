import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import RecipeCatalogCard from '../../../components/common/RecipeCatalogCard';

describe('RecipeCatalogCard component', () => {
  it('should render component', () => {
    const props = {
      user: {
        isAuthenticated: true,
      },
      recipe: {
        id: 1,
        title: 'Beans',
        category: 'Main',
        description: 'Tasty beans',
        preparationTime: 85,
        ingredients: 'Beans, onions, palm oil, salt',
        directions: 'Do this. Do that',
        views: 34,
        upvotes: 20,
        downvotes: 48,
        favorites: 5,
        User: {
          id: 1,
          fullName: 'John Mark',
        },
      },
    };
    const wrapper = mount(
      <BrowserRouter><RecipeCatalogCard {...props} /></BrowserRouter>
    );
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should notify users to signin before viewing other users profile', () => {
    const props = {
      user: {
        isAuthenticated: false,
      },
      recipe: {
        id: 1,
        title: 'Beans',
        category: 'Main',
        description: 'Tasty beans',
        preparationTime: 85,
        ingredients: 'Beans, onions, palm oil, salt',
        directions: 'Do this. Do that',
        views: 34,
        upvotes: 20,
        downvotes: 48,
        favorites: 5,
        User: {
          id: 1,
          fullName: 'John Mark',
        },
      },
    };
    const wrapper = mount(
      <BrowserRouter><RecipeCatalogCard {...props} /></BrowserRouter>
    );

    expect(wrapper).toBeDefined();
    wrapper.find('a#view-user-profile').simulate('click', {
      preventDefault: jest.fn(),
    });
    expect(wrapper).toMatchSnapshot();
  });
});
