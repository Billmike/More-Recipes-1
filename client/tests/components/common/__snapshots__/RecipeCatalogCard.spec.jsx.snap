// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCatalogCard component should notify users to signin before viewing other users profile 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RecipeCatalogCard
      recipe={
        Object {
          "User": Object {
            "fullName": "John Mark",
            "id": 1,
          },
          "category": "Main",
          "description": "Tasty beans",
          "directions": "Do this. Do that",
          "downvotes": 48,
          "favorites": 5,
          "id": 1,
          "ingredients": "Beans, onions, palm oil, salt",
          "preparationTime": 85,
          "title": "Beans",
          "upvotes": 20,
          "views": 34,
        }
      }
      user={
        Object {
          "isAuthenticated": false,
        }
      }
    >
      <div
        className="col s12 m4 l3"
      >
        <div
          className="card large hoverable"
          id="recipes"
        >
          <Link
            replace={false}
            to="recipes/1"
          >
            <a
              href="recipes/1"
              onClick={[Function]}
            >
              <span
                className="card-title"
              >
                Beans
              </span>
            </a>
          </Link>
          <div
            className="card-image"
          >
            <Link
              replace={false}
              to="recipes/1"
            >
              <a
                href="recipes/1"
                onClick={[Function]}
              >
                <RecipeImage
                  recipeImage={null}
                  title="Beans"
                >
                  <img
                    alt="Beans"
                    src={null}
                  />
                </RecipeImage>
              </a>
            </Link>
          </div>
          <div
            className="card-content"
          >
            <p>
              Tasty beans
            </p>
          </div>
          <div
            className="card-action"
          >
            <p
              id="owner"
            >
              <label
                htmlFor="owner"
              >
                Recipe by: 
              </label>
              <a
                href="#!"
                id="view-user-profile"
                onClick={[Function]}
              >
                <span
                  className="recipe-owner"
                >
                  John Mark
                </span>
              </a>
            </p>
            <ul
              className="center-align"
            >
              <li
                id="views"
              >
                <i
                  className="material-icons tiny"
                >
                  visibility
                </i>
                34
              </li>
              <li
                id="favorites"
              >
                <i
                  className="material-icons tiny"
                >
                  favorite
                </i>
                5
              </li>
              <li
                id="upvotes"
              >
                <i
                  className="material-icons tiny"
                >
                  thumb_up
                </i>
                20
              </li>
              <li
                id="downvotes"
              >
                <i
                  className="material-icons tiny"
                >
                  thumb_down
                </i>
                48
              </li>
            </ul>
          </div>
        </div>
      </div>
    </RecipeCatalogCard>
  </Router>
</BrowserRouter>
`;

exports[`RecipeCatalogCard component should render component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RecipeCatalogCard
      recipe={
        Object {
          "User": Object {
            "fullName": "John Mark",
            "id": 1,
          },
          "category": "Main",
          "description": "Tasty beans",
          "directions": "Do this. Do that",
          "downvotes": 48,
          "favorites": 5,
          "id": 1,
          "ingredients": "Beans, onions, palm oil, salt",
          "preparationTime": 85,
          "title": "Beans",
          "upvotes": 20,
          "views": 34,
        }
      }
      user={
        Object {
          "isAuthenticated": true,
        }
      }
    >
      <div
        className="col s12 m4 l3"
      >
        <div
          className="card large hoverable"
          id="recipes"
        >
          <Link
            replace={false}
            to="recipes/1"
          >
            <a
              href="recipes/1"
              onClick={[Function]}
            >
              <span
                className="card-title"
              >
                Beans
              </span>
            </a>
          </Link>
          <div
            className="card-image"
          >
            <Link
              replace={false}
              to="recipes/1"
            >
              <a
                href="recipes/1"
                onClick={[Function]}
              >
                <RecipeImage
                  recipeImage={null}
                  title="Beans"
                >
                  <img
                    alt="Beans"
                    src={null}
                  />
                </RecipeImage>
              </a>
            </Link>
          </div>
          <div
            className="card-content"
          >
            <p>
              Tasty beans
            </p>
          </div>
          <div
            className="card-action"
          >
            <p
              id="owner"
            >
              <label
                htmlFor="owner"
              >
                Recipe by: 
              </label>
              <Link
                replace={false}
                to="/users/1/dashboard"
              >
                <a
                  href="/users/1/dashboard"
                  onClick={[Function]}
                >
                  <span
                    className="recipe-owner"
                  >
                    John Mark
                  </span>
                </a>
              </Link>
            </p>
            <ul
              className="center-align"
            >
              <li
                id="views"
              >
                <i
                  className="material-icons tiny"
                >
                  visibility
                </i>
                34
              </li>
              <li
                id="favorites"
              >
                <i
                  className="material-icons tiny"
                >
                  favorite
                </i>
                5
              </li>
              <li
                id="upvotes"
              >
                <i
                  className="material-icons tiny"
                >
                  thumb_up
                </i>
                20
              </li>
              <li
                id="downvotes"
              >
                <i
                  className="material-icons tiny"
                >
                  thumb_down
                </i>
                48
              </li>
            </ul>
          </div>
        </div>
      </div>
    </RecipeCatalogCard>
  </Router>
</BrowserRouter>
`;
